src/Analyzer.Core.BuiltInRuleTests/README.md

```diff
@@ -3,46 +3,35 @@ This project is designed for quickly adding tests to verify the correctness of b
+To create a set of tests for a rule, a new directory and JSON file are created.  **The name of the directory and file must be the same as the `id` property of the JSON rule**, with ".json" as the file extension.
 
-For example, a JSON rule like the following:
-``` js
-{
-    "name": "SuperSecurityCheck",
-    "description": "...",
-    "recommendation": "...",
-    "helpUri": "...",
-    "evaluation": {
-      ...
-    }
-}
-```
-... would have tests defined in a configuration file named **"Tests/SuperSecurityCheck.json"**.
+For example, to write tests for rule ID *TA-000001*, a test configuration file must be located at *TA-000001/TA-000001.json* within the *Tests* directory.
 
 The JSON test configuration has the following schema:
 ``` js
 [
     {
-        "Template": "Name of template file analyzed (without file extension).  Template must be in the 'TestTemplates' directory.",
+        "Template": "Name of template file analyzed.  Template must be within the same directory as the test configuration file.",
         "ReportedFailures": [ // Array of objects with integer line numbers - each are a line number expected to be reported in the failure.
             {
                 "LineNumber": 3, // Line number of expected reported failure
                 "Description": "(Optional) Description of what's being tested for this expected failure."
             }
+        // Any other "made up" properties can be added as well, as the test author deems appropriate; for example, providing context on why some resources are expected to pass analysis.
         ]
     },
     ... // More tests can be defined if multiple templates should be analyzed - one test block for each template
 ]
 ```
 
-### Test ARM Templates
-For each template file referenced in a `Template` property of a test configuration, there should be a file in the *TestTemplates* directory with the same name, **having ".badtemplate" as the file extension**.  (This extension is used to help prevent these templates from actually being deployed in Azure.)
+Although `Template`, `ReportedFailures`, and `LineNumber` are required properties, the test config is not limited to having only these properties.  If test authors choose, other properties can be created simply by adding them in the JSON file.  This can be helpful for giving additional context, for example to explain why certain resources in a test template do not fail (for testing that the rule does not generate false-positives).
 
-For example, if the value of `Template` is "SuperSecurityCheck_failure", there is expected to be a template file at path *TestTemplates/SuperSecurityCheck_failure.badtemplate*.
+### Test ARM Templates
+For each template file referenced in a `Template` property of a test configuration, there should be a file within the same directory with the same name.
```

Comment:
This part does not have the example. Do you want to continue the example? At least for me, the "SuperSecurityCheck" example is how I understood what was needed.

Reply:
Added the example back.